js/models/message_handler.js

```diff
@@ -155,6 +155,27 @@ export default class MessageHandler {
     this.aladin.select(selectionType);
   }
 
+  handleTriggerSelectionByTable(msg) {
+    let sources = msg["table"];
+    let keys = msg["keys"];
+    let sourcesByCatalog = this.aladin.view.catalogs
+      .map((cat) => {
+        if (!cat.isShowing) {
+          return;
+        }
+        return cat.getSources().filter((source) =>
+          sources.some((s) =>
+            keys.every((key) => {
+              return s[key] == source.data[key];
+            }),
+          ),
+        );
+      })
+      .filter((n) => n.length);
```

Comment:
note: this filter statement is here to remove any empty lists that may be returned from the first filter statement. If there are multiple catalogs, and the key only matches on some of them, it will return something like the following
```
[
    [],     // empty list
    [ ... ] // list full of sources
]
```
The empty list needs to be removed for the `selectObjects` call to function properly

Reply:
I feel like there is already enough happening with all the nested functions in the `map()` line.  I would just do this last `filter()` on a new line preceded by a comment.